examples/userapi/03_subdomains.ipynb

```diff
@@ -935,14 +906,73 @@
   },
   {
    "cell_type": "code",
-   "execution_count": 35,
+   "execution_count": 32,
    "metadata": {},
    "outputs": [],
    "source": [
     "from devito import norm\n",
     "assert np.isclose(norm(v[0]), 0.10301, rtol=1e-4)"
    ]
   },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Legacy API\n",
+    "\n",
+    "The legacy `SubDomain` API required `SubDomain`s to be created prior to the `Grid`, and passed to the `Grid` on instantiation. This interface can still be used to avoid breaking backward compatibility with existing codes. We can use the `Middle`, `Left`, and `Right` classes from earlier to illustrate this API."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 33,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stderr",
+     "output_type": "stream",
+     "text": [
+      "Operator `Kernel` ran in 0.01 s\n"
+     ]
+    },
+    {
+     "data": {
+      "text/plain": [
+       "Data([[2., 2., 2., 2., 2., 2., 2., 2., 5., 5.],\n",
+       "      [2., 2., 2., 2., 2., 2., 2., 2., 5., 5.],\n",
+       "      [0., 0., 0., 0., 0., 0., 0., 0., 3., 3.],\n",
+       "      [0., 0., 0., 0., 1., 1., 1., 0., 3., 3.],\n",
+       "      [0., 0., 0., 0., 1., 1., 1., 0., 3., 3.],\n",
+       "      [0., 0., 0., 0., 1., 1., 1., 0., 3., 3.],\n",
+       "      [0., 0., 0., 0., 0., 0., 0., 0., 3., 3.],\n",
+       "      [0., 0., 0., 0., 0., 0., 0., 0., 3., 3.],\n",
+       "      [0., 0., 0., 0., 0., 0., 0., 0., 3., 3.],\n",
+       "      [0., 0., 0., 0., 0., 0., 0., 0., 3., 3.]], dtype=float32)"
+      ]
+     },
+     "execution_count": 33,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "left = Left()\n",
+    "right = Right()\n",
+    "mid = Middle()\n",
+    "\n",
+    "new_grid = Grid(shape=(10, 10), subdomains=(left, right, mid))\n",
+    "\n",
+    "g = Function(name='g', grid=new_grid)\n",
+    "\n",
+    "eq1 = Eq(g, g+1, subdomain=new_grid.subdomains['middle'])\n",
```

Comment:
IIRC you can just pass `subdomain=mid`, and same story below `subdomain=left`

That makes it a bit neater, at least

I know that all tests in `test_subdomain.py` use this format, but for example check out other tests such as those here https://github.com/devitocodes/devito/blob/master/tests/test_gpu_common.py#L208

Reply:
You can do that, but I wanted to keep it true to the original examples